account_asset_disposal/models/account_move_line.py

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Antonio Espinosa <antonio.espinosa@tecnativa.com>
+# Copyright 2017 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# Copyright 2017 Luis M. Ontalba - <luis.martinez@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, models
+from odoo.exceptions import Warning as UserError
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    @api.multi
+    def unlink(self):
+        disposed_lines = self.env['account.asset.asset'].search([]).filtered(
```

Comment:
You shouldn't check this (as this is very performance drainer), and disposal move should be treated the same as the rest of the moves created from asset (it's inserted also in the depreciation board): if Odoo allows to manually remove them, then let it do it.